src/common/transactions/transaction-utils.ts

```diff
@@ -47,6 +57,7 @@ export const generateContractCallTx = ({
     postConditions: getPostConditions(txData.postConditions),
     network,
     sponsored,
+    ...(parsedFeeValue ? { fee: parsedFeeValue } : {}),
```

Comment:
I'd suggest removing this, I can integrate the changes into my PR, there are a few other places this will need to change